### ISSUE TITLE
WARNING - NaN or Inf found in input tensor.

### ISSUE THREAD

**Issue by 1304150468** [80chen86/IPDN#7]
在`train_gres.sh`的训练过程中，出现以下警告，这是正常现象吗?
```
2025-06-28 09:58:56,642 - INFO - Epoch [1/70][700/2860]  lr: 0.0001, eta: 2 days, 22:37:54, data_time: 0.01, iter_time: 1.27, score_loss: 0.0000, mask_bce_loss: nan, mask_dice_loss: nan, sem_loss: nan, indi_loss: nan, sample_loss: nan, layer_0_score_loss: 0.0000, layer_0_mask_bce_loss: nan, layer_0_mask_dice_loss: nan, layer_0_sem_loss: nan, layer_0_indi_loss: nan, layer_1_score_loss: 0.0000, layer_1_mask_bce_loss: nan, layer_1_mask_dice_loss: nan, layer_1_sem_loss: nan, layer_1_indi_loss: nan, layer_2_score_loss: 0.0000, layer_2_mask_bce_loss: nan, layer_2_mask_dice_loss: nan, layer_2_sem_loss: nan, layer_2_indi_loss: nan, layer_3_score_loss: 0.0000, layer_3_mask_bce_loss: nan, layer_3_mask_dice_loss: nan, layer_3_sem_loss: nan, layer_3_indi_loss: nan, layer_4_score_loss: 0.0000, layer_4_mask_bce_loss: nan, layer_4_mask_dice_loss: nan, layer_4_sem_loss: nan, layer_4_indi_loss: nan, layer_5_score_loss: 0.0000, layer_5_mask_bce_loss: nan, layer_5_mask_dice_loss: nan, layer_5_sem_loss: nan, layer_5_indi_loss: nan, loss: nan, grad_total_norm: nan
2025-06-28 09:59:09,874 - WARNING - NaN or Inf found in input tensor.
2025-06-28 09:59:09,875 - WARNING - NaN or Inf found in input tensor.
2025-06-28 09:59:09,875 - WARNING - NaN or Inf found in input tensor.
```
经查找好像是`dec.py`里的迭代过程中因为没有`>0.75`的值导致权重为nan，能说一下可能是什么原因吗？
```
# multi-round
l = 0
for i in range(self.num_layer):
    #PAD
    lang_query, w1, w2 = self.lqg[i](query[:,:128],lang_feats)
    if i > l:
        mask = pred_indis.softmax(-1)[:,:,1]>0.75
        weight = (w2 * (~init_lang_mask).unsqueeze(-1)).sum(1)
        weight = torch.masked_fill(weight,~mask,float('-inf')).softmax(-1)
```

**Comment by songchuanle-1** [comment]
I also meet this problem , but i  have not followed code :
# multi-round
l = 0
for i in range(self.num_layer):
    #PAD
    lang_query, w1, w2 = self.lqg[i](query[:,:128],lang_feats)
    if i > l:
        mask = pred_indis.softmax(-1)[:,:,1]>0.75
        weight = (w2 * (~init_lang_mask).unsqueeze(-1)).sum(1)
        weight = torch.masked_fill(weight,~mask,float('-inf')).softmax(-1)

NaN or Inf found in input tensor  will present in random epoch......

**Comment by songchuanle-1** [comment]
> 在`train_gres.sh`的训练过程中，出现以下警告，这是正常现象吗?
> 
> ```
> 2025-06-28 09:58:56,642 - INFO - Epoch [1/70][700/2860]  lr: 0.0001, eta: 2 days, 22:37:54, data_time: 0.01, iter_time: 1.27, score_loss: 0.0000, mask_bce_loss: nan, mask_dice_loss: nan, sem_loss: nan, indi_loss: nan, sample_loss: nan, layer_0_score_loss: 0.0000, layer_0_mask_bce_loss: nan, layer_0_mask_dice_loss: nan, layer_0_sem_loss: nan, layer_0_indi_loss: nan, layer_1_score_loss: 0.0000, layer_1_mask_bce_loss: nan, layer_1_mask_dice_loss: nan, layer_1_sem_loss: nan, layer_1_indi_loss: nan, layer_2_score_loss: 0.0000, layer_2_mask_bce_loss: nan, layer_2_mask_dice_loss: nan, layer_2_sem_loss: nan, layer_2_indi_loss: nan, layer_3_score_loss: 0.0000, layer_3_mask_bce_loss: nan, layer_3_mask_dice_loss: nan, layer_3_sem_loss: nan, layer_3_indi_loss: nan, layer_4_score_loss: 0.0000, layer_4_mask_bce_loss: nan, layer_4_mask_dice_loss: nan, layer_4_sem_loss: nan, layer_4_indi_loss: nan, layer_5_score_loss: 0.0000, layer_5_mask_bce_loss: nan, layer_5_mask_dice_loss: nan, layer_5_sem_loss: nan, layer_5_indi_loss: nan, loss: nan, grad_total_norm: nan
> 2025-06-28 09:59:09,874 - WARNING - NaN or Inf found in input tensor.
> 2025-06-28 09:59:09,875 - WARNING - NaN or Inf found in input tensor.
> 2025-06-28 09:59:09,875 - WARNING - NaN or Inf found in input tensor.
> ```
> 
> 经查找好像是`dec.py`里的迭代过程中因为没有`>0.75`的值导致权重为nan，能说一下可能是什么原因吗？
> 
> ```
> # multi-round
> l = 0
> for i in range(self.num_layer):
>     #PAD
>     lang_query, w1, w2 = self.lqg[i](query[:,:128],lang_feats)
>     if i > l:
>         mask = pred_indis.softmax(-1)[:,:,1]>0.75
>         weight = (w2 * (~init_lang_mask).unsqueeze(-1)).sum(1)
>         weight = torch.masked_fill(weight,~mask,float('-inf')).softmax(-1)
> ```

我也遇到了这个问题，， 他会随机出现在不同的epoch中    这是由于损失变成了None导致的，但是不知道因为设么原因导致的 ，，，，我把你上述的代码注释进行训练也会遇到相同问题

**Comment by 1304150468** [comment]
可以检查一下是不是`dec.py`中部分的`proj_tokens`被改成了`proj_queries`，比如
```
proj_queries = []
if self.contrastive_align_loss:
    proj_queries.append(F.normalize(self.contrastive_align_projection_vision(query), p=2, dim=-1))
else:
    proj_queries.append(None)
    
proj_tokens = []
if self.contrastive_align_loss:
    # proj_queries.append(F.normalize(self.contrastive_align_projection_vision(query), p=2, dim=-1)) # 原来错误的
    proj_tokens.append(F.normalize(self.contrastive_align_projection_vision(query), p=2, dim=-1)) # 这样是对的
else:
    proj_tokens.append(None)
```
反正我是这个原因。。。
